manifests/init.pp

```diff
@@ -1,49 +1,91 @@
-#
-# [*package_name*]
-#   Only valid when the install_method == package. Defaults to `nomad`.
-#
-# [*package_ensure*]
-#   Only valid when the install_method == package. Defaults to `installed`.
-#
-#
-# [*restart_on_change*]
-#   Determines whether to restart nomad agent on $config_hash changes.
-#   This will not affect reloads when service, check or watch configs change.
-# Defaults to `true`.
+# @example To set up a single nomad server, with several agents attached, on the server.
+#   class { '::nomad':
+#     version     => '1.0.1', # check latest version at https://github.com/hashicorp/nomad/blob/master/CHANGELOG.md
+#     config_hash => {
+#       'region'     => 'us-west',
+#       'datacenter' => 'ptk',
+#       'log_level'  => 'INFO',
+#       'bind_addr'  => '0.0.0.0',
+#       'data_dir'   => '/opt/nomad',
+#       'server'     => {
+#         'enabled'          => true,
+#         'bootstrap_expect' => 3,
+#       }
+#     }
+#   }
 #
-# [*extra_options*]
-#   Extra arguments to be passed to the nomad agent
+# @example On the agent(s)
+#   class { 'nomad':
+#     config_hash => {
+#         'region'     => 'us-west',
+#         'datacenter' => 'ptk',
+#         'log_level'  => 'INFO',
+#         'bind_addr'  => '0.0.0.0',
+#         'data_dir'   => '/opt/nomad',
+#         'client'     => {
+#         'enabled'    => true,
+#         'servers'    => [
+#           "nomad01.your-org.pvt:4647",
+#           "nomad02.your-org.pvt:4647",
+#           "nomad03.your-org.pvt:4647"
+#         ]
+#       }
+#     },
+#  }
 #
-# [*init_style*]
-#   What style of init system your system uses.
+# @example Disable install and service components
+#   class { '::nomad':
+#     install_method => 'none',
+#     init_style     => false,
+#     manage_service => false,
+#     config_hash   => {
+#       'region'     => 'us-west',
+#       'datacenter' => 'ptk',
+#       'log_level'  => 'INFO',
+#       'bind_addr'  => '0.0.0.0',
+#       'data_dir'   => '/opt/nomad',
+#       'client'     => {
+#       'enabled'    => true,
+#       'servers'    => [
+#           "nomad01.your-org.pvt:4647",
+#           "nomad02.your-org.pvt:4647",
+#           "nomad03.your-org.pvt:4647"
+#         ]
+#       }
+#     },
+#  }
 #
-# [*purge_config_dir*]
-#   Purge config files no longer generated by Puppet
+# @param arch cpu architecture
+# @param manage_user manage the user that will run nomad
+# @param user username
+# @param manage_group manage the group that will run nomad
+# @param extra_groups additional groups to add the nomad user
+# @param purge_config_dir Purge config files no longer generated by Puppet
+# @param group groupname
+# @param join_wan join nomad cluster over the WAN
+# @param bin_dir location of the nomad binary
+# @param version Specify version of nomad binary to download.
+# @param install_method install via system package, download and extract from a url. Defaults to `url` or disable install.
```

Comment:
I think you can remove `defaults to` since the reference automatically extracts it. Same below

Reply:
removed them